src/model.rs

```diff
@@ -777,6 +777,39 @@ impl Collections {
         }
         self.vehicle_journeys = CollectionWithId::new(vehicle_journeys).unwrap();
     }
+
+    /// Many calendars are identicall and can be deduplicate
+    pub fn calendar_deduplication(&mut self) {
+        fn find_duplicate_calendar<'a>(
+            calendars: &'a [Calendar],
+            dates: &BTreeSet<NaiveDate>,
+        ) -> Option<&'a Calendar> {
+            for c in calendars {
+                if c.dates == *dates {
+                    return Some(c);
+                }
+            }
+            None
+        }
+
+        let mut calendars_used: Vec<Calendar> = vec![];
+        let mut vehicle_journeys = self.vehicle_journeys.take();
+        vehicle_journeys.sort_unstable_by_key(|vj| vj.id.clone());
```

Comment:
I'm not sure I understand why you need to sort the `vehicle_journeys` by `id`? `vehicle_journeys` and `calendars` are both stored in `CollectionWithId` which are already ordered and the order is not gonna be altered (only some elements are gonna be removed, but relative positions are gonna be the same). Or am I missing something?

If I am missing something and the sorting needs to stay, please consider the following solution to avoid the `.clone()` operation.
```suggestion
        vehicle_journeys.sort_unstable_by(|vj_a, vj_b| { 
            vj_a.service_id                              
                .partial_cmp(&vj_b.service_id)           
                .unwrap_or(std::cmp::Ordering::Equal)    
        });                                              
```

Reply:
why not `vehicle_journeys.sort_unstable_by(|vj1, vj2| vj1.service_id.cmp(&vj2.service_id));` ?

Reply:
Yes, why not? That's obviously better than mine! No reason, just didn't look deep enough I guess.